schema/binary.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Namespace Labs Inc; All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+
+package schema
+
+import (
+	"google.golang.org/protobuf/proto"
+	"namespacelabs.dev/foundation/internal/fnerrors"
+)
+
+// Modifies the instance in place.
+func (bc *BinaryConfig) MergeFrom(other *BinaryConfig) error {
```

Comment:
I'd like to avoid OO for protos. Protos should be exclusively data, and attaching behavior to them makes their API surface cover two areas, and makes it more difficult to reason about e.g. versioning.

We have a few existing helper methods for protos, that cast to domain-specific types (e.g. string to PackageName) which are OK because they don't really implement any other behavior beyond type casting.

You'll see from my comments above that I got confused by the use of `MergeFrom` as I assumed you were using proto's Merge methods.

Lets keep these type of methods in a package that makes sense for them, so its clearly readable what the intended behavior is, e.g. `deploy.CombineBinaryConfig()`, or `planning.CombineBinaryConfig`.